### PR TITLE
Fix for DCOS-10699

### PR DESCRIPTION
**Pull request by GoelDeepak**

### CODE REVIEW

**Comment by GoelDeepak** [comment]
@aeyakovenko @nlsun please review

File: rebar.config [hunk 1]
```diff
@@ -20,7 +20,7 @@
     {erlzk, {git, "https://github.com/huaban/erlzk.git", {tag, "v0.6.2"}}},
     {gpb, {git, "https://github.com/tomas-abrahamsson/gpb.git", {tag, "3.21.3"}}},
     {gen_netlink, ".*", {git, "https://github.com/mesosphere/gen_netlink.git", {branch, "master"}}},
-    {minuteman, {git, "https://github.com/dcos/minuteman.git", master}}
+    {minuteman, {git, "https://github.com/GoelDeepak/minuteman.git", {branch, "dev"}}}
```
Comment (nlsun) [inline rebar.config]: This should be changed back

File: rebar.lock [hunk 2]
```diff
@@ -87,8 +87,8 @@
        {ref,"7fa7e6edb4d2eecc0ee12a1f109accddf1978953"}},
   0},
  {<<"minuteman">>,
-  {git,"https://github.com/dcos/minuteman.git",
-       {ref,"452b57b07d08def9a4586b6be046943785440529"}},
+  {git,"https://github.com/GoelDeepak/minuteman.git",
```
Comment (nlsun) [inline rebar.lock]: This should be changed back

File: apps/navstar_dns/src/navstar_dns_listener.erl [hunk 3]
```diff
@@ -215,7 +219,6 @@ push_signed_zone(PublicKey, _Zone0 = {ZoneName0, _ZoneSha, Records0}) ->
     case rpc:call(spartan_name(), erldns_zone_cache, put_zone, [Zone1], ?RPC_TIMEOUT) of
         Reason = {badrpc, _} ->
             lager:warning("Unable to push signed records to spartan: ~p", [Reason]),
-            setup_retry(ZoneName0),
```
Comment (nlsun) [inline apps/navstar_dns/src/navstar_dns_listener.erl:218]: Why did you remove the retry here?

File: apps/navstar_dns/src/navstar_dns_listener.erl [hunk 4]
```diff
@@ -184,27 +186,29 @@ retry_spartan(Key = [navstar, dns, zones, ZoneName]) ->
...
-            setup_retry(ZoneName),
             ok;
         ok ->
             maybe_push_signed_zone(Zone),
             ok
     end.
 
-setup_retry(ZoneName) ->
-    {ok, _} = timer:send_after(?SPARTAN_RETRY, {retry_spartan, [navstar, dns, zones, ZoneName]}).
+setup_retry() ->
```
Comment (nlsun) [inline apps/navstar_dns/src/navstar_dns_listener.erl:199]: Maybe we want 2 types of retry? 1 args for partial, no args for full retry?

File: apps/navstar_dns/src/navstar_dns_listener.erl [hunk 5]
```diff
@@ -12,7 +12,7 @@
 -behaviour(gen_server).
 
 %% API
--export([start_link/0, convert_zone/3]).
+-export([start_link/0, convert_zone/3, retry_spartan/1]).
```
Comment (nlsun) [inline apps/navstar_dns/src/navstar_dns_listener.erl]: Why did you export retry_spartan?

File: apps/navstar_dns/src/navstar_dns_listener.erl (hunk 4 above)
Comment (GoelDeepak) [reply apps/navstar_dns/src/navstar_dns_listener.erl:199]: do we need partial one when the full one is every 30 sec?

File: apps/navstar_dns/src/navstar_dns_listener.erl (hunk 3 above)
Comment (GoelDeepak) [reply apps/navstar_dns/src/navstar_dns_listener.erl:218]: to have it as part of full retry

File: rebar.config (hunk 1 above)
Comment (GoelDeepak) [reply rebar.config]: will do once minuteman change is committed

File: rebar.lock (hunk 2 above)
Comment (GoelDeepak) [reply rebar.lock]: yes

File: apps/navstar_dns/src/navstar_dns_listener.erl (hunk 5 above)
Comment (GoelDeepak) [reply apps/navstar_dns/src/navstar_dns_listener.erl]: Can we still call it from lists:foreach?

Comment (nlsun) [reply apps/navstar_dns/src/navstar_dns_listener.erl]: I think you might only need to export it if you're calling it with the `module:fun()` format? Although I'd have to check.

File: apps/navstar_dns/src/navstar_dns_listener.erl (hunk 4 above)
Comment (nlsun) [reply apps/navstar_dns/src/navstar_dns_listener.erl:199]: Hmm I see, that makes sense.

**Comment by aeyakovenko** [comment]
can you add a CT test for this too
